src/main/java/gregtech/common/items/tool/EntityDamageBehavior.java

```diff
@@ -0,0 +1,56 @@
+package gregtech.common.items.tool;
+
+import gregtech.api.damagesources.DamageSources;
+import gregtech.api.items.toolitem.behavior.IToolBehavior;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.DamageSource;
+import net.minecraft.world.World;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+public class EntityDamageBehavior implements IToolBehavior {
+
+    private final List<Function<EntityLivingBase, Float>> shouldDoBonusList = new ArrayList<>();
+    private final String mobType;
```

Comment:
Is this designed in a way to be really variable or Minecraft does not have some base mob class which has name on it, from which everyone inherits?